### ISSUE TITLE
The children utility function causes the child components to initialize in advance.

### ISSUE THREAD

**Issue by siaikin** [solidjs/solid#1944]
### Describe the bug

I use Context to inject some data globally, and I optimize the access to children according to the [examples](https://www.solidjs.com/tutorial/props_children) given in the tutorial, that is, using the children utility function to wrap `props.children`. However, the child components always access the `initialValue` of `createContext` through `useContext`, instead of the data I provide through Provider.

### Your Example Website or App

https://playground.solidjs.com/anonymous/29fa1b7c-efd7-4f32-a6f7-1a37c22b047c

### Steps to Reproduce the Bug or Issue

1. Go to https://playground.solidjs.com/anonymous/29fa1b7c-efd7-4f32-a6f7-1a37c22b047c
2. Button render `-1`, this is `initialValue`.
3. Please comment out this `IS NOT WORK` code, 16-19 lines.
4. Than comment in `IS WORK` code, 22-24 lines.
5. Butten Render `-2`, this is provide by `Context.Provider`.

### Expected behavior

I hope that when using the `children` utility function, the child components will not be initialized in advance, causing `useContext` to be unable to get the provided value.

### Platform

- OS: Windows
- Browser: Chrome

**Comment by titoBouzout** [comment]
See this https://github.com/solidjs/solid/issues/1926
TLDR: using `children` changes the context on which things run, so children isn't evaluated inside the context as you may expect.

**Comment by siaikin** [comment]
> See this #1926 TLDR: using `children` changes the context on which things run, so children isn't evaluated inside the context as you may expect.

very grateful for your answer, I spent several hours troubleshooting this issue. Perhaps we could add explanations to the documentation or tutorials to explain this hidden behavior.

**Comment by ryansolid** [comment]
Yeah docs could be clearer here with a context example. We say that it resolves the children so they can be introspected, but for those not familiar with that Solid doesn't use a VDOM so resolved means do the rendering it isn't obvious this can happen. We use conditional rendering as an example but not context. https://www.solidjs.com/docs/latest/api#children
